### PR TITLE
fix: trims fname/lname on save and dob -> numbers

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/4023)

- [x] This change addresses the issue in full

## Description
We ran into a few issues caused by two things
1. first name and last name of applicants and houseHoldMembers sometimes had a ` ` trailing. This trims existing data and also prevents this in the future by trimming before writing to db

2. birth year, birth month, and birth day fields were stored as strings which meant that '01' != '1'. This converts those fields to be numbers in the db (along with saving them as numbers, and converting existing data to numbers)

## How Can This Be Tested/Reviewed?
unit and e2e tests were updated to reflect these changes

but on the front end if you apply to a listing and put a name like `fname ` as the first name, `lname ` as the last name, `01` as birth month, `02` as birth day then the data should get saved in the db (and displayed on the front end) as `fname`, `lname`, `1`, `2` 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: api/test/unit/services/application-flagged-set.service.spec.ts [hunk 1]
```diff
@@ -134,23 +134,15 @@ describe('Testing application flagged set service', () => {
               householdMember: {
                 some: {
                   birthMonth: {
-                    in: [
-                      'example birth month',
-                      'household birth moth 1',
-                      'household birth moth 2',
```
Comment (mcgarrye) [inline api/test/unit/services/application-flagged-set.service.spec.ts:140]: lol moth

**Review by mcgarrye — COMMENTED** [review]
Code changes look good. Question: For the migration, do we want to cleanup AFS ruleKeys 1. by trimming first/last names and 2. removing leading zeros from birthday and birthmonth?

File: api/test/integration/application-flagged-set.e2e-spec.ts [hunk 2]
```diff
@@ -60,7 +60,7 @@ describe('Application flagged set Controller Tests', () => {
 
   const createComplexApplication = async (
     emailIndicator: string,
-    nameAndDOBIndicator: string,
+    nameAndDOBIndicator: number,
     listing: string,
     dobIndicator?: string,
```
Comment (mcgarrye) [inline api/test/integration/application-flagged-set.e2e-spec.ts:65]: Can we change this to nameIndicator since that's how you are using it?

Comment (YazeedLoonat) [reply api/test/integration/application-flagged-set.e2e-spec.ts:65]: updated!

**Review by mcgarrye — APPROVED** [review]

**Pull request merged**